src/device.rs

```diff
@@ -125,9 +135,9 @@ pub unsafe extern "C" fn wgpuAdapterGetProperties(
     adapter: native::WGPUAdapter,
     properties: &mut native::WGPUAdapterProperties,
 ) {
-    let adapter = adapter.expect("invalid adapter");
+    let (adapter, context) = unwrap_context_handle(adapter);
 
-    let maybe_props = gfx_select!(adapter => GLOBAL.adapter_get_info(adapter));
+    let maybe_props = gfx_select!(adapter => context.adapter_get_info(adapter));
     if let Ok(props) = maybe_props {
         properties.name = ADAPTER_NAMES
```

Comment:
we can remove `ADAPTER_NAMES` global and instead return CString ptr and mem::forget it, making it user's responsibility to free the name string.

Reply:
I moved the cached string into the adapter, could use mem::forget too but it hasn't been used like this (as a member of a returned structure) so I wanted to avoid that.

Reply:
make sense, seems like [dawn does](https://source.chromium.org/chromium/chromium/src/+/main:third_party/dawn/src/dawn/native/Adapter.cpp;l=89;drc=af07b3e7e43bd39dc707d11c3e32e12606c1d3bb) something similar.
Though we will need to add more strings later (description, architecture)